### PR TITLE
TEC-3238 Hook into filter in common/Utils/Body_Classes.

### PR DESCRIPTION
**Pull request by Camwyn**
Requires: https://github.com/moderntribe/tribe-common/pull/1385

https://moderntribe.atlassian.net/browse/TEC-3238

### CODE REVIEW

File: src/Tribe/Views/V2/Hooks.php [hunk 1]
```diff
@@ -299,13 +301,25 @@ public function filter_map_canonical_query_args( $map, $url, $rewrite ) {
...
 	 */
-	public function filter_body_class( $classes ) {
-		$classes = $this->container->make( Theme_Compatibility::class )->filter_add_body_classes( $classes );
+	public function filter_body_classes( $classes ) {
 		$classes = $this->container->make( Template_Bootstrap::class )->filter_add_body_classes( $classes );
 
 		return $classes;
 	}
 
+	public function add_body_classes() {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php:310]: :no_entry_sign: Missing doc comment for function add_body_classes()

<sub>hash: 9b1861e2ca063d2bef6384dbf15c6024</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 2]
```diff
@@ -299,13 +301,25 @@ public function filter_map_canonical_query_args( $map, $url, $rewrite ) {
...
+		$classes = $this->container->make( Theme_Compatibility::class )->add_body_classes();
+		$classes = $this->container->make( Template_Bootstrap::class )->add_body_classes();
+	}
+
+	public function body_classes_should_add( $add, $queue ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php:315]: :no_entry_sign: Missing doc comment for function body_classes_should_add()

<sub>hash: 1128aa8364e8a571b3693c58849eb252</sub>

File: src/Tribe/Views/V2/Template_Bootstrap.php [hunk 3]
```diff
@@ -371,23 +372,31 @@ public function filter_add_body_classes( $classes ) {
...
+				unset( $classes[ $key ] );
 			}
-		} else {
-			$classes[] = 'tribe-events-page-template';
 		}
 
 		return $classes;
 	}
 
+	public function add_body_classes() {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Template_Bootstrap.php:392]: :no_entry_sign: Missing doc comment for function add_body_classes()

<sub>hash: 84a87817188a372f4ec387d0ec0db391</sub>

File: src/Tribe/Views/V2/Template_Bootstrap.php [hunk 4]
```diff
@@ -371,23 +372,31 @@ public function filter_add_body_classes( $classes ) {
...
+		$setting  = $this->get_template_setting();
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Template_Bootstrap.php:393]: :warning: Equals sign not aligned correctly; expected 1 space but found 2 spaces

<sub>hash: 9365f10dc3830d7965585be61e2fec21</sub>

File: src/Tribe/Views/V2/Theme_Compatibility.php [hunk 5]
```diff
@@ -75,6 +77,18 @@ public function filter_add_body_classes( array $classes ) {
 		return array_merge( $classes, $this->get_body_classes() );
 	}
 
+	public function add_body_classes() {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Theme_Compatibility.php:80]: :no_entry_sign: Missing doc comment for function add_body_classes()

<sub>hash: 41ce5aeee356964850f3e5fb60dc97d7</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 6]
```diff
@@ -299,13 +301,41 @@ public function filter_map_canonical_query_args( $map, $url, $rewrite ) {
...
+	}
+
+	/**
+	 * Logic for if body classes should be added.
+	 *
+	 * @since TBD
+	 *
+	 * @param boolean $add   Whether to add classes or not.
+	 * @param string  $queue The queue we want to get 'admin', 'display', 'all'.
+	 * @return void
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php:329]: :no_entry_sign: Function return type is void, but function contains return statement

<sub>hash: e88b8faaae992163125af1c0b385ad3f</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 7]
```diff
@@ -299,13 +301,48 @@ public function filter_map_canonical_query_args( $map, $url, $rewrite ) {
...
+	 *
+	 * @since TBD
+	 *
+	 * @param boolean $add   Whether to add classes or not.
+	 * @param string  $queue The queue we want to get 'admin', 'display', 'all'.
+	 *
+	 * @return boolean Whether to add classes or not.
+	 */
+	public function body_classes_should_add( $add, $queue ) {
+		$context = tribe_context();
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php:333]: :warning: Equals sign not aligned with surrounding assignments; expected 6 spaces but found 1 space

<sub>hash: 3ea15802f3dec764fa6c069c1ce7deb1</sub>

File: src/Tribe/Template/Single_Event.php [hunk 8]
```diff
@@ -119,5 +121,24 @@ public function set_notices() {
...
+
+		/**
+		 * Hook into filter and add our logic for adding body classes.
+		 *
+		 * @since TBD
+		 *
+		 * @param boolean $add              Whether to add classes or not.
+		 * @param string  $queue            The queue we want to get 'admin', 'display', 'all'.
+		 *
+		 * @return boolean
```
Comment (lucatume) [inline src/Tribe/Template/Single_Event.php:133]:
```suggestion
		 * @return boolean Whether body classes should be added or not.
```

File: src/Tribe/Views/V2/Hooks.php [hunk 9]
```diff
@@ -299,13 +302,65 @@ public function filter_map_canonical_query_args( $map, $url, $rewrite ) {
...
+	 * @since TBD
+	 *
+	 * @param boolean $add   Whether to add the class to the queue or not.
+	 * @param array   $class The array of body class names to add.
+	 * @param string  $queue The queue we want to get 'admin', 'display', 'all'.
+	 * @return boolean
+	 */
+	public function body_class_should_add_to_queue( $add, $class, $queue ) {
+		$add = $this->container->make( Template_Bootstrap::class )->should_add_body_class_to_queue( $add, $class, $queue );
+
```
Comment (lucatume) [inline src/Tribe/Views/V2/Hooks.php:335]: Remove this blank line?

File: src/Tribe/Views/V2/Template_Bootstrap.php [hunk 10]
```diff
@@ -371,23 +372,60 @@ public function filter_add_body_classes( $classes ) {
...
 
+	/**
+	 * Contains the logic for if this object's classes should be added to the queue.
+	 *
+	 * @since TBD
+	 *
+	 * @param boolean $add   Whether to add the class to the queue or not.
+	 * @param array   $class The array of body class names to add.
+	 * @param string  $queue The queue we want to get 'admin', 'display', 'all'.
+	 * @return boolean
```
Comment (lucatume) [inline src/Tribe/Views/V2/Template_Bootstrap.php:400]:
```suggestion

	 * @return boolean Whether body classes should be added or not.
```

File: src/Tribe/Views/V2/Theme_Compatibility.php [hunk 11]
```diff
@@ -75,6 +78,43 @@ public function filter_add_body_classes( array $classes ) {
...
 
+	/**
+	 * Contains the logic for if this object's classes should be added to the queue.
+	 *
+	 * @since TBD
+	 *
+	 * @param boolean $add   Whether to add the class to the queue or not.
+	 * @param array   $class The array of body class names to add.
+	 * @param string  $queue The queue we want to get 'admin', 'display', 'all'.
+	 * @return boolean
```
Comment (lucatume) [inline src/Tribe/Views/V2/Theme_Compatibility.php:89]:
```suggestion

	 * @return boolean Whether body classes should be added or not.
```

File: src/Tribe/Views/V2/Hooks.php (hunk 9 above)
Comment (Camwyn) [reply src/Tribe/Views/V2/Hooks.php:335]:
```suggestion
```

**Pull request merged**